libpldm/pdr.c

```diff
@@ -162,9 +162,11 @@ static pldm_pdr_record *make_new_record(const pldm_pdr *repo,
 		 * caller supplied the record handle, it would exist in the
 		 * header already.
 		 */
-		struct pldm_pdr_hdr *hdr =
-		    (struct pldm_pdr_hdr *)(record->data);
-		hdr->record_handle = htole32(record->record_handle);
+		if (!record_handle) {
```

Comment:
This was changed on purpose to solve this - https://github.com/ibm-openbmc/pldm/pull/143/
Bringing back this would again cause the same problem.